avaframe/com1DFA/DFAfunctionsCython.pyx

```diff
@@ -1298,11 +1329,11 @@ def computeGradC(cfg, particles, headerNeighbourGrid, headerNormalGrid, double[:
                   if r < rKernel:
                       hr = rKernel - r
                       wKernel = facKernel * hr * hr * hr
-                      m11 = m11 + mass[l] / h[l] * wKernel * r1 * r1 / rho
-                      m12 = m12 + mass[l] / h[l] * wKernel * r1 * r2 / rho
-                      m22 = m22 + mass[l] / h[l] * wKernel * r2 * r2 / rho
-                      G1 = G1 + mass[l] * (1 - h[j]/h[l]) * wKernel * K1*r1
-                      G2 = G2 + mass[l] * (1 - h[j]/h[l]) * wKernel * K2*r2
+                      m11 = m11 + mass[l] / hArray[l] * wKernel * r1 * r1 / rho
+                      m12 = m12 + mass[l] / hArray[l] * wKernel * r1 * r2 / rho
+                      m22 = m22 + mass[l] / hArray[l] * wKernel * r2 * r2 / rho
```

Comment:
This, and the line above is basically the same code as above (line 1309ff). I suggest making a function out of it.

Reply:
his is @matthiasto fucntion and I dont understand everything. I do prefer updqte this in q next PR or let him do